### ISSUE TITLE
Open Source Driver Station Organization?

### ISSUE THREAD

**Issue by gluxon** [anidev/frc-driverstation#13]
Hello everyone!

I started looking at creating an [open source driver station](https://github.com/gluxon/DriverStation.js) a couple years back. Since then, many other talented coders have independently gained much more ground on the project.

Due to that, I would like to pitch the creation of a GitHub organization we can all merge our projects and collaborate under. I acknowledge that the work I've done is far less than the work in this repository, so I encourage whoever is leading this repository to make a decision that I hope is free from my pressure.

Brandon

**Comment by codyps** [comment]
Even just having a list of them in a README of a repo (perhaps just for the purpose of containing the list) would be useful.

**Comment by LoadingPleaseWait** [comment]
That sounds like a great idea! I could not agree with you more!

**Comment by lopsided98** [comment]
I think the best use of that type of a common organization would be to document the protocol. Right now, different parts of it are scattered over all these threads and projects.

**Comment by anidev** [comment]
I agree, I think a central location for collaborating would be a great idea. We could definitely use a central location for documenting the current status of reversing and documenting the protocol. A list with links to existing projects would be a good idea too. However I feel like this alone is more suited for a wiki than a code repository. Github does have a wiki capability, but we would have to create a code repository for that. We could put common files such as protocol dissectors (if that's fine with you @lopsided98) and packet dumps in there, but I can't think of much else, so I'm not sure if Github is best suited for this purpose. I think that in addition to the wiki we might want something to help with collaboration and keeping everyone who is interested up to date (as opposed to comments on Github issues). That's open for debate, might be overkill. For sure I think we should have a wiki somewhere with current progress on the protocol.

As for merging all the individual projects under one umbrella, I'm not so sure about. Each project is distinct, with developers who probably have their own ideas for how to manage their project and which direction they might want to take it in. We would have to migrate our individual code bases to the organization and work out things like contributor access. Maybe I'm taking this too seriously; that's also open for discussion. For now, a list of current projects is definitely a good thing to have.

So for now that's a wiki with protocol documentation and a wiki or README with links to current projects. Whether Github is the best platform for this should be considered. Thoughts?

**Comment by lopsided98** [comment]
I agree that a wiki system is probably the best format. I also agree that it doesn't seem like the best idea for everyone to migrate their code into one organization. It would be nice if GitHub had a way to group projects with different owners under one organization.

I probably won't be able to maintain the Android app that well once I graduate, so at that point I might consider moving ownership to an organization.

I'd be happy to put my Wireshark dissectors in a more accessible location, but we would still have to figure out contributors and ownership. It might be easier to just create a wiki (possibly not on GitHub) with links to all relevant resources and projects, as well as protocol documentation.

**Comment by alex-spataru** [comment]
Sorry for coming late to discussion, I have created a [repository](https://github.com/alex-spataru/DS-Database) which contains a list of the open-source DriverStation projects that I found (both 2015 and 2014). However, I also agree that a wiki would be the best option to list the projects and (possibly) document the DS protocols over the years.

**Comment by lopsided98** [comment]
@alex-spataru That's a good list, I didn't know about a few of them, especially QDriverStation, which looks pretty functional.

**Comment by lopsided98** [comment]
@alex-spataru There are a couple things I noticed, though:
- [Android DriverStation](https://github.com/raystubbs/Android-FRC-Driverstation), should be credited to Ray Stubbs (team 2657), not me
- RobotDS should be RoboDS

**Comment by alex-spataru** [comment]
@lopsided98 Thanks for the feedback, I fixed my mistakes in the file.
